src/module.c

```diff
@@ -2737,6 +2737,35 @@ double RM_LoadDouble(RedisModuleIO *io) {
     return value;
 }
 
+/* Load an encoded data type from a RedisModuleString.  This enables implementing
+ * a RESTORE like command. */
+void *RM_LoadDataTypeFromString(const RedisModuleString *str, const moduleType *mt) {
```

Comment:
I'd rename it something like `RM_ModuleTypeRestoreValue` and the other one `RM_ModuleTypeDumpValue`